xl-v1-multipart.go

```diff
@@ -631,9 +631,11 @@ func (xl xlObjects) CompleteMultipartUpload(bucket string, object string, upload
 
 	// Rename if an object already exists to temporary location.
 	uniqueID := getUUID()
-	err = xl.renameObject(bucket, object, minioMetaBucket, path.Join(tmpMetaPrefix, uniqueID))
-	if err != nil {
-		return "", toObjectErr(err, bucket, object)
+	if xl.isObject(bucket, object) {
+		err = xl.renameObject(bucket, object, minioMetaBucket, path.Join(tmpMetaPrefix, uniqueID))
```

Comment:
```
diff --git a/xl-v1-object.go b/xl-v1-object.go
index de0a427..b825055 100644
--- a/xl-v1-object.go
+++ b/xl-v1-object.go
@@ -171,6 +171,9 @@ func (xl xlObjects) getObjectInfo(bucket, object string) (ob
 // its proper location.
 func (xl xlObjects) renameObject(srcBucket, srcObject, dstBucket, dstObject str
        // Initialize sync waitgroup.
+       if !xl.isObject(srcBucket, srcObject) {
+               return nil
+       }
        var wg = &sync.WaitGroup{}
 
        // Initialize list of errors.
```

I made the above change and it is causing an issue with xl multipart uploads. After any multipart upload, a GET request is resulting in 404 status code.